### PR TITLE
Disable cd / dvd repositories when installation is over

### PR DESCRIPTION
**Pull request by mchf**
Fate#325541

Depends on https://github.com/yast/yast-installation-control/pull/72

### CODE REVIEW

File: test/pkg_finish_test.rb [hunk 1]
```diff
@@ -77,19 +80,24 @@
...
 
     context "given some local repository" do
-      let(:repositories) { [local_repo, remote_repo] }
+      let(:repositories) { [local_repo, local_dvd_repo, remote_repo] }
       let(:base_products) { [sles_product, sled_product] }
 
       let(:local_repo) do
         Y2Packager::Repository.new(repo_id: 1, name: "SLE-12-SP2-0", enabled: true,
-          url: URI("cd://dev/sr0"), autorefresh: false)
+          url: URI("dir:///path/to/extracted_ISO"), autorefresh: false)
```
Comment (lslezak) [inline test/pkg_finish_test.rb]: The `dir://` schema makes only sense in installed system, you cannot use it for installing a system (`/` is in inst-sys in RAM disk).

But you can use the `hd`or `iso` schema for installation, like `hd:/dir?device=/dev/sda1`. For details see "Supported URI formats" section in `man zypper`.

File: src/lib/packager/clients/pkg_finish.rb [hunk 2]
```diff
@@ -102,6 +102,19 @@ def write
...
+    # Checks if given source repo should be disabled
+    #
+    # Currently CD / DVD repositories are disabled after installation, when
+    # enabled in control file
+    #
+    # @param source [Symbol] if :cd or :dvd then control file is checked if the
+    #                        repo should be disabled
+    # @return [Boolean] true if the repo should be disabled
+    def disable_media_repo?(source)
+      disable_repo = ProductFeatures.GetBooleanFeature("software", "disable_media_repo")
```
Comment (lslezak) [inline src/lib/packager/clients/pkg_finish.rb]: Just an optimization, we do not have to evaluate the `disable_media_repo` option if the source is not CD or DVD.

```ruby
return false unless [:cd, :dvd].include?(source)
ProductFeatures.GetBooleanFeature("software", "disable_media_repo")
```

File: test/pkg_finish_test.rb (hunk 1 above)
Comment (mchf) [reply test/pkg_finish_test.rb]: well, I picked the first one which came to my mind and was not cd/dvd ... but you're right, I'll do it more "real world" example

File: src/lib/packager/clients/pkg_finish.rb (hunk 2 above)
Comment (mchf) [reply src/lib/packager/clients/pkg_finish.rb]: ok

**Pull request merged**